### PR TITLE
Saved queries new UI

### PR DESCRIPTION
**Pull request by amsiglan**
### Description
This PR updates the UI for saved queries to use Flyouts which give more surface to search and categorize the queries as well as introduce more options when saving them.

## Screenshot
- Saved queries action popup
image

- Save query action item when clicked opens
image

- Open query action item when clicked opens
image

- Templates tab (Note: these queries cannot be deleted from UI) 
image

## Testing the changes
Tested change with local setup.

## Changelog
- feat: Enhances the saved query UX

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+              {/* <EuiFlexItem grow={false}>
+                <EuiCheckbox
+                  id="run-saved-query-automatically"
+                  checked={shouldRunOnOpen}
+                  onChange={(event) => {
+                    setShouldRunOnOpen(event.target.checked);
+                  }}
+                  label={'Run automatically'}
+                />
+              </EuiFlexItem> */}
```
Comment (amsiglan) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:185]: This commented code is supposed to be uncommented once we have the logic in place to control running of query on open

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  return (
+    <EuiFlyout onClose={onClose} hideCloseButton>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle>
+          <h3>Saved queries</h3>
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiSearchBar
+          box={{
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:114]:
```suggestion
        <EuiSearchBar
          compressed
          box={{
```

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+              multiSelect: 'or',
+              options: Array.from(queryLanguageSet).map((language) => ({
+                value: language,
+                view: language.toUpperCase(),
+              })),
+            },
+          ]}
+          onChange={onChange}
+        />
+        <EuiSpacer />
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:133]:
```suggestion
        <EuiSpacer size="s" />
```

@kgcreative can you please sign off on this suggestion?

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+              key={query.id}
+              savedQuery={queryMap[query.id].savedQuery}
+              selectedQuery={selectedQuery}
+              onSelect={setSelectedQuery}
+              onRunPreview={onQueryOpen}
+              onClose={onClose}
+            />
+          ))
+        ) : (
+          <EuiEmptyPrompt content={'No saved query present'} />
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:146]: Please use i18n.

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+  );
+
+  pager.current.setTotalItems(filteredSavedQueries.length);
+  const queriesOnCurrentPage = filteredSavedQueries.slice(firstItemIndex, lastItemIndex + 1);
+
+  return (
+    <EuiFlyout onClose={onClose} hideCloseButton>
+      <EuiFlyoutHeader hasBorder>
+        <EuiTitle>
+          <h3>Saved queries</h3>
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:109]: Please use i18n

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        </EuiTitle>
+      </EuiFlyoutHeader>
+      <EuiFlyoutBody>
+        <EuiSearchBar
+          box={{
+            placeholder: 'Search saved query',
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:115]: i18n plz

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+            incremental: true,
+            schema,
+          }}
+          filters={[
+            {
+              type: 'field_value_selection',
+              field: 'language',
+              name: 'Query language',
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:123]: i18n plz

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+          <EuiFlexItem grow={false}>
+            <EuiFlexGroup alignItems="center">
+              {/* <EuiFlexItem grow={false}>
+                <EuiCheckbox
+                  id="run-saved-query-automatically"
+                  checked={shouldRunOnOpen}
+                  onChange={(event) => {
+                    setShouldRunOnOpen(event.target.checked);
+                  }}
+                  label={'Run automatically'}
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:183]: i18n plz

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+            setFirstItemIndex(pager.current.getFirstItemIndex());
+            setLastItemIndex(pager.current.getLastItemIndex());
+          }}
+        />
+      </EuiFlyoutBody>
+      <EuiFlyoutFooter>
+        <EuiFlexGroup gutterSize="s" justifyContent="spaceBetween">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty iconType={'cross'} color="danger" iconSide="left" onClick={onClose}>
+              Cancel
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:171]: i18n plz

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+                  disabled={!selectedQuery}
+                  fill
+                  onClick={() => {
+                    if (selectedQuery) {
+                      onQueryOpen(selectedQuery);
+                      onClose();
+                    }
+                  }}
+                >
+                  Open query
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:197]: i18n plz

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+            </div>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false} style={{ marginTop: 0 }}>
+            <EuiCopy textToCopy={savedQuery.attributes.query.query as string}>
+              {(copy) => (
+                <EuiButtonIcon
+                  onClick={copy}
+                  iconType="copy"
+                  iconSize="m"
+                  aria-label="Copy recent query"
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx:168]: i18n plz

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss [hunk 12]
```diff
@@ -0,0 +1,30 @@
+.editor-container {
+  position: relative;
+  overflow-y: clip;
+}
+
+.read-more-wrap {
+  /* stylelint-disable  @osd/stylelint/no_restricted_values */
+  background: linear-gradient(rgba($euiColorEmptyShade, 5%) 10%, $euiColorEmptyShade 60%, $euiColorEmptyShade 100%);
+  height: 90px;
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss:9]: Can u plz add a comment about why 10% and 60%.
Also why 90px?

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+  left: 0;
+  bottom: 0;
+  text-align: center;
+  width: 100%;
+}
+
+.read-more-btn {
+  position: absolute;
+  left: 50%;
+  bottom: 13px;
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss:21]: If this is required, please figure out the 13px based on OUI variables.

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+  transform: translateX(-50%);
+}
+
+.copy-saved-query-btn {
+  position: absolute;
+  right: 5px;
+  top: -5px;
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss:28]: Please use OUI variables

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+  font-size: 20px;
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss:29]: Why are we overriding OUI's typography?

File: src/plugins/data/public/ui/saved_query_form/helpers.tsx [hunk 16]
```diff
@@ -0,0 +1,312 @@
...
+  // a saved query with a time filter will override whatever the current value of the global timepicker
+  // is. We expect this option to be used rarely and only when the user knows they want this behavior.
+  const [shouldIncludeTimefilter, setIncludeTimefilter] = useState(false);
+  const [formErrors, setFormErrors] = useState<string[]>([]);
+
+  // Instead of setting the default values based on the saved query above,
+  // we use this effect so that in case when user select "Save as new" in the flyout,
+  // the initial state resets with change to 'savedQuery'
+  useEffect(() => {
+    setTitle(savedQuery ? savedQuery.title : '');
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_form/helpers.tsx:93]: Wouldn't it be safer to do this?
```suggestion
    setTitle(savedQuery?.title || '');
```

File: src/plugins/data/public/ui/saved_query_form/helpers.tsx [hunk 17]
```diff
@@ -0,0 +1,312 @@
...
+    setEnabledSaveButton(Boolean(savedQuery));
+    setDescription(savedQuery ? savedQuery.description : '');
+    setShouldIncludeFilters(savedQuery ? !!savedQuery.filters : true);
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_form/helpers.tsx:96]: nit
```suggestion
    setShouldIncludeFilters(!!savedQuery?.filters);
```

File: src/plugins/data/public/ui/saved_query_form/helpers.tsx [hunk 18]
```diff
@@ -0,0 +1,312 @@
...
+  const [shouldIncludeTimefilter, setIncludeTimefilter] = useState(false);
+  const [formErrors, setFormErrors] = useState<string[]>([]);
+
+  // Instead of setting the default values based on the saved query above,
+  // we use this effect so that in case when user select "Save as new" in the flyout,
+  // the initial state resets with change to 'savedQuery'
+  useEffect(() => {
+    setTitle(savedQuery ? savedQuery.title : '');
+    setEnabledSaveButton(Boolean(savedQuery));
+    setDescription(savedQuery ? savedQuery.description : '');
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_form/helpers.tsx:95]: if `savedQuery` is defined and `savedQuery.description` is undefined, we will have a problem. right?
```suggestion
    setDescription(savedQuery?.description || '');
```

File: src/plugins/data/public/ui/saved_query_form/helpers.tsx [hunk 19]
```diff
@@ -0,0 +1,312 @@
...
+    setShouldIncludeFilters(savedQuery ? !!savedQuery.filters : true);
+    setIncludeTimefilter(savedQuery ? !!savedQuery.timefilter : false);
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_form/helpers.tsx:97]: nit:
```suggestion
    setIncludeTimefilter(!!savedQuery?.timefilter);
```

File: src/plugins/data/public/ui/saved_query_form/helpers.tsx [hunk 20]
```diff
@@ -0,0 +1,312 @@
...
+    fetchQueries();
+  }, [savedQueryService]);
+
+  const validate = useCallback(() => {
+    const errors = [];
+    if (
+      !!savedQueries.find(
+        (existingSavedQuery) => !savedQuery && existingSavedQuery.attributes.title === title
+      )
+    ) {
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_form/helpers.tsx:130]:
```suggestion
    if (
      !savedQuery && 
      savedQueries.some(
        (existingSavedQuery) => existingSavedQuery.attributes?.title === title
      )
    ) {
```

Comment (AMoo-Miki) [reply src/plugins/data/public/ui/saved_query_form/helpers.tsx:130]: Are we checking title elsewhere?

File: src/plugins/data/public/ui/saved_query_form/helpers.tsx [hunk 21]
```diff
@@ -0,0 +1,312 @@
...
+          {savedQueryDescriptionText}
+        </EuiText>
+      </EuiCompressedFormRow>
+      <EuiCompressedFormRow
+        label={i18n.translate('data.search.searchBar.savedQueryNameLabelText', {
+          defaultMessage: 'Name',
+        })}
+        helpText={i18n.translate('data.search.searchBar.savedQueryNameHelpText', {
+          defaultMessage:
+            'Name is required. Name cannot contain leading or trailing whitespace. Name must be unique.',
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_form/helpers.tsx:181]:
```suggestion
            'Name is required and must be unique, without leading or trailing spaces.',
```

File: src/plugins/data/public/ui/saved_query_form/helpers.tsx [hunk 22]
```diff
@@ -0,0 +1,312 @@
...
+                >
+                  Save as new
+                </EuiButton>
+              </EuiFlexItem>
+            )}
+            <EuiFlexItem grow={false}>
+              <EuiButton fill disabled={hasErrors || !enabledSaveButton} onClick={onClickSave}>
+                {i18n.translate('data.search.searchBar.savedQueryFormSaveButtonText', {
+                  defaultMessage: savedQuery ? 'Save changes' : 'Save',
+                })}
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_form/helpers.tsx:291]: i18n cannot be dynamic like this. Please break into two different translates.

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx [hunk 23]
```diff
@@ -186,7 +204,50 @@ export function SavedQueryManagementComponent({
...
 
-  return (
+  return useNewSavedQueryUI ? (
+    <div
+      className="osdSavedQueryManagement__popover"
+      data-test-subj="saved-query-management-popover"
+    >
+      <EuiListGroup>
+        <EuiListGroupItem
+          label={'Save query'}
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:214]: i18n plz

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx [hunk 24]
```diff
@@ -186,7 +204,50 @@ export function SavedQueryManagementComponent({
...
+          iconType={'save'}
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:215]: For readability:
```suggestion
          iconType="save"
```

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx [hunk 25]
```diff
@@ -186,7 +204,50 @@ export function SavedQueryManagementComponent({
...
+                  showFilterOption={true}
+                  showTimeFilterOption={true}
+                  savedQuery={loadedSavedQuery?.attributes}
+                />
+              )
+            );
+          }}
+        />
+        <EuiListGroupItem
+          label={'Open query'}
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:233]: i18n plz

File: src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx [hunk 26]
```diff
@@ -186,7 +204,50 @@ export function SavedQueryManagementComponent({
...
+          iconType={'folderOpen'}
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx:234]: for readability:
```suggestion
          iconType="folderOpen"
```

File: src/plugins/data/public/ui/search_bar/search_bar.tsx [hunk 27]
```diff
@@ -332,7 +332,7 @@ class SearchBarUI extends Component<SearchBarProps, State> {
       if (this.props.onSaved) {
         this.props.onSaved(response);
       }
-    } catch (error) {
+    } catch (error: any) {
       this.services.notifications.toasts.addDanger(
         `An error occured while saving your query: ${error.message}`
```
Comment (AMoo-Miki) [inline src/plugins/data/public/ui/search_bar/search_bar.tsx:337]: While at it, it would be great if you could add i18n here

File: yarn.lock [hunk 28]
```diff
@@ -15981,7 +15981,7 @@ string-similarity@^4.0.1:
   resolved "https://registry.yarnpkg.com/string-similarity/-/string-similarity-4.0.4.tgz#42d01ab0b34660ea8a018da8f56a3309bb8b2a5b"
   integrity sha512-/q/8Q4Bl4ZKAPjj8WerIBJWALKkaPRfrvhfF8k/B23i4nzrlRj2/go1m90In7nG/3XDSbOo0+pu6RvCTM9RGMQ==
 
-"string-width-cjs@npm:string-width@^4.2.0":
+"string-width-cjs@npm:string-width@^4.2.0", string-width@^4.1.0, string-width@^4.2.0, string-width@^4.2.3:
```
Comment (AMoo-Miki) [inline yarn.lock:15984]: Plz revert these changes

File: yarn.lock
Comment (AMoo-Miki) [inline yarn.lock:1]: Please revert all of these changes made to the lockfile.

**Comment by ashwin-pc** [comment]
Hey @amsiglan can you change the folder icon that triggers the popover to use `boxesHorizontal` instead?

File: src/plugins/data/public/ui/saved_query_form/helpers.tsx (hunk 20 above)
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_form/helpers.tsx:132]: Yes, if title is undefined the save button stays disabled

File: src/plugins/data/public/ui/search_bar/search_bar.tsx (hunk 27 above)
Comment (amsiglan) [reply src/plugins/data/public/ui/search_bar/search_bar.tsx]: done

File: yarn.lock
Comment (amsiglan) [reply yarn.lock:1]: done

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss (hunk 14 above)
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss]: Removed this

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss (hunk 15 above)
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/saved_query_flyouts.scss]: Removed this

**Review by ashwin-pc — COMMENTED** [review]
Didnt get a chance to review the whole thing yet. But a quick question on saved query templates. I see you added the tab, but I couldnt find a way in the UI to create them. Also if our plan is to include them dynamically, can we hide the tab if there arent any present? Having an empty tab there for most users is bad UX

File: src/plugins/data/public/query/saved_query/saved_query_service.ts [hunk 29]
```diff
@@ -33,34 +33,43 @@ import { SavedObjectsClientContract, SavedObjectAttributes, CoreStart } from 'sr
...
+  const queryEnhancementEnabled = uiSettings.get(UI_SETTINGS.QUERY_ENHANCEMENTS_ENABLED);
+
   const saveQuery = async (attributes: SavedQueryAttributes, { overwrite = false } = {}) => {
     if (!attributes.title.length) {
       // title is required extra check against circumventing the front end
       throw new Error('Cannot create saved query without a title');
     }
 
-    const query = {
+    const query: any = {
```
[the remark this answers is not in the document]
Comment (amsiglan) [reply src/plugins/data/public/query/saved_query/saved_query_service.ts]: fixed

**Comment by amsiglan** [comment]
> Didnt get a chance to review the whole thing yet. But a quick question on saved query templates. I see you added the tab, but I couldnt find a way in the UI to create them. Also if our plan is to include them dynamically, can we hide the tab if there arent any present? Having an empty tab there for most users is bad UX

Will follow up with another PR to update the UI with either:
- Add a way to support template creation OR
- Hide the template tab when there are no templates

Another angle is that now since we will always have some sample queries that we show where there are no results, we could package them as templates as well. 

Will work on the best path forward and create a follow up PR

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx (hunk 1 above)
Comment (virajsanghvi) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:266]: Ideally we could just add this later, but if we're including it, can we leave a comment in the code for why its commented out?

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx [hunk 30]
```diff
@@ -0,0 +1,199 @@
...
+}: SavedQueryCardProps) {
+  const [shouldTruncate, setShouldTruncate] = useState(false);
+  const [isTruncated, setIsTruncated] = useState(true);
+  const [editorHeight, setEditorHeight] = useState(60);
+  const customHTMLRef = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
+  const [showDeletionConfirmationModal, setShowDeletionConfirmationModal] = useState(false);
+
+  const [lineCount, setLineCount] = useState(0);
+
+  // Function to toggle the view state
```
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx:46]: nit: probably can drop this comment

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx [hunk 31]
```diff
@@ -0,0 +1,199 @@
...
+                    color="danger"
+                    onClick={() => {
+                      setShowDeletionConfirmationModal(true);
+                    }}
+                  />
+                </EuiFlexItem>
+              )}
+            </EuiFlexGroup>
+            {savedQuery.attributes.description && (
+              <EuiText>
```
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx:125]: please use small on EuiText elements

File: src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx [hunk 32]
```diff
@@ -0,0 +1,199 @@
...
+                editorDidMount={handleHTMLEditorDidMount}
+              />
+              {shouldTruncate && (
+                <div className={isTruncated ? 'read-more-wrap' : ''}>
+                  <EuiLink className="read-more-btn" onClick={toggleView}>
+                    {i18n.translate('data.saved_query.view_more_label', {
+                      defaultMessage: '{viewMoreLabel}',
+                      values: {
+                        viewMoreLabel: `${
+                          isTruncated ? `View full query (${lineCount}) lines` : 'View Less'
```
Comment (virajsanghvi) [inline src/plugins/data/public/ui/saved_query_flyouts/saved_query_card.tsx:159]: should lines be in the parens?

**Review by virajsanghvi — COMMENTED** [review]
There are tests failing related to saved queries.

Also, we've added a lot of code without tests - how are we tracking following up with those tests?

File: src/plugins/data/public/ui/search_bar/search_bar.tsx [hunk 33]
```diff
@@ -334,9 +342,15 @@ class SearchBarUI extends Component<SearchBarProps, State> {
...
         this.props.onSaved(response);
       }
-    } catch (error) {
+    } catch (error: any) {
       this.services.notifications.toasts.addDanger(
-        `An error occured while saving your query: ${error.message}`
+        this.props.intl.formatMessage(
+          {
+            id: 'data.search_bar.save_query.failedToSaveQuery',
+            defaultMessage: 'An error occured while saving your query: {errorMessage}',
```
[the remark this answers is not in the document]
Comment (amsiglan) [reply src/plugins/data/public/ui/search_bar/search_bar.tsx:350]: Handled the undefined case. For type annotation either it can be `any` or `unknown`, typescript doesn't allow specific type for error in catch clause

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 34]
```diff
@@ -0,0 +1,287 @@
...
+  savedQuery: SavedQuery;
+}
+
+export function OpenSavedQueryFlyout({
+  savedQueryService,
+  onClose,
+  onQueryOpen,
+  handleQueryDelete,
+}: OpenSavedQueryFlyoutProps) {
+  // const [shouldRunOnOpen, setShouldRunOnOpen] = useState(false);
```
[the remark this answers is not in the document]
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:50]: This is supposed to be used in a follow up PR from another team member, but I guess will remove it for now

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 35]
```diff
@@ -0,0 +1,287 @@
...
+    setQueriesOnCurrentPage(
+      filteredSavedQueries.slice(
+        pager.current.getFirstItemIndex(),
+        pager.current.getLastItemIndex() + 1
+      )
+    );
+  }, [savedQueries, searchQuery, activePage, itemsPerPage]);
+
+  const onChange: EuiSearchBarProps['onChange'] = ({ query, error }) => {
+    if (!error) {
```
[the remark this answers is not in the document]
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:123]: It will show no results which is fine since user will simply need to change the search query

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 36]
```diff
@@ -0,0 +1,287 @@
...
+  };
+
+  const flyoutBodyContent = (
+    <>
+      <EuiSpacer />
+      <EuiSearchBar
+        compressed
+        query={searchQuery}
+        box={{
+          placeholder: 'Search saved query',
```
[the remark this answers is not in the document]
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:151]: fixed in next iteration

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx (hunk 1 above)
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx]: Removed for now

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx (hunk 34 above)
[the remark this answers is not in the document]
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx]: Removed for now

**Comment by amsiglan** [comment]
> There are tests failing related to saved queries.
> 
> Also, we've added a lot of code without tests - how are we tracking following up with those tests?

Let me add a task on myself to add new tests

**Review by virajsanghvi — DISMISSED** [review]
I did not have time to pull this down to validate, and looks like there are still some more cases of text to be localized, but approving assuming that and unit tests will be followed up with.

**Comment by ananzh** [comment]
So we have both save queries when we click showFilterActions button and the new one to save queries? What is the purpose or usage pain to resolve for adding another save query?

**Comment by amsiglan** [comment]
> So we have two places 1) click showFilterActions button and 2) the new three dot one to save queries? What is the purpose or usage pain to resolve for adding another save query? are we going to remove the old one to make the new one the only place to save query?
> 
> Screenshot 2024-10-24 at 12 53 34 PM

We are replacing the old button in both the places, the one inside the filter options will also show the two options
- Save query and Open query

**Comment by ananzh** [comment]
> > So we have two places 1) click showFilterActions button and 2) the new three dot one to save queries? What is the purpose or usage pain to resolve for adding another save query? are we going to remove the old one to make the new one the only place to save query?
> > Screenshot 2024-10-24 at 12 53 34 PM
> 
> We are replacing the old button in both the places, the one inside the filter options will also show the two options
> 
> * Save query and Open query

Got it. Haven't got time to review it in details, only verify the changes. Since you have resolve multiple people's comments, I will approve to unblock. Will reach out to you there is any concerns in the future.

**Review by ananzh — DISMISSED** [review]

File: src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx [hunk 37]
```diff
@@ -0,0 +1,283 @@
...
+            defaultMessage: 'Search',
+          }),
+          incremental: true,
+          schema,
+        }}
+        filters={[
+          {
+            type: 'field_value_selection',
+            field: 'datasetType',
+            name: 'Data type',
```
[the remark this answers is not in the document]
Comment (amsiglan) [reply src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx:161]: Updating in follow up PR

**Review by virajsanghvi — APPROVED** [review]

**Review by ruanyl — APPROVED** [review]

**Comment by ashwin-pc** [comment]
Rerunning CI Group 6 since the associated fix has been merged. Will merge once its passed

**Comment by AMoo-Miki** [comment]
manual backport was merged into 2.x
